### PR TITLE
4 create business index view

### LINKED ISSUE #4: Users can see businesses
As a user (right now, regardless of role),
 I visit the business index page
I see links to individual businesses

### PR DESCRIPTION
**Pull request by seeker105**
Add `business_id` field to Items table. Create 'Business Index' link on navbar. Create Business index page. Each business name is a link to an index of that business's items.

closes #4

### CODE REVIEW

File: app/views/businesses/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<% if @businesses.count > 0 %>
```
Comment (ErinGreenhalgh) [inline app/views/businesses/index.html.erb:1]: I think there is a more active record way to do this, something like if businesses.exist... we can look into it

File: app/assets/javascripts/businesses.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
```
Comment (saylerb) [inline app/assets/javascripts/businesses.coffee:2]: We could probably delete these automatically generated files.

File: spec/factories/business_admins.rb [hunk 3]
```diff
@@ -1,6 +0,0 @@
-FactoryGirl.define do
```
Comment (ErinGreenhalgh) [inline spec/factories/business_admins.rb]: let's delete this file

File: spec/integration/user_can_visit_business_index_page_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.feature "user can view business index" do
+  scenario "user visits home page and sees the business index link" do
+    visit root_url
+
+    expect(page).to have_link("Business Index")
```
Comment (ErinGreenhalgh) [inline spec/integration/user_can_visit_business_index_page_spec.rb:7]: let's add a scope to this: within navbar or whatever the class we would expect is

File: spec/integration/user_can_visit_business_index_page_spec.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  scenario "user clicks 'Business Index' link on home page" do
+    businesses = create_list(:business, 3)
+    visit root_url
+
+    click_link "Business Index"
+    expect(current_path).to eq(businesses_path)
```
Comment (ErinGreenhalgh) [inline spec/integration/user_can_visit_business_index_page_spec.rb:15]: let's hardcode in the path: expect(path).to eq('/"#business.name"') 
so we can make sure the vanity route is displaying the way we want

File: spec/integration/user_can_visit_business_index_page_spec.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  scenario "user clicks on a link to be taken to a business show page" do
+    businesses = create_list(:business, 2)
+    business1 = businesses[0]
+    items = create_list(:item, 3)
+    business1.items << items
+
+    visit businesses_path
+
+    click_link(business1.name)
+    expect(current_path).to eq(business_path(id: business1.id))
```
Comment (ErinGreenhalgh) [inline spec/integration/user_can_visit_business_index_page_spec.rb:30]: again let's hard code a path that we expect

**Comment by saylerb** [comment]
Let's merge this after we make some of the fixes that Erin suggested...

File: spec/integration/user_can_visit_business_index_page_spec.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  scenario "user clicks 'Business Index' link on home page" do
+    businesses = create_list(:business, 3)
+    visit root_url
+
+    click_link "Business Index"
+    expect(current_path).to eq(businesses_path)
+    expect(page).to have_link(businesses[0].name)
+    expect(page).to have_link(businesses[1].name)
+    expect(page).to have_link(businesses[2].name)
+  end
```
Comment (ErinGreenhalgh) [inline spec/integration/user_can_visit_business_index_page_spec.rb:19]: add scopes to all these assertions

**Pull request merged**